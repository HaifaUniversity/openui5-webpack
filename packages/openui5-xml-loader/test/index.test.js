import path from 'path';
import loader from '../src/index';

describe('XML Loader', () => {
  it('should handle custom rootPaths', (done) => {
    const xml = `<mvc:View xmlns:mvc="sap.ui.core.mvc" xmlns:c="my.custom.namespace">
  <c:TestControl />
</mvc:View>`;

    const addDependency = jest.fn();

    loader.apply({
      query: {
        modulePaths: {
          'my.custom.namespace': path.resolve(__dirname, 'fixtures/custom')
        }
      },
      async: () => (err, data) => {
        expect(data).toMatchSnapshot();
        done();
      },
      context: __dirname,
      addDependency,
    }, [xml]);
  })
});
