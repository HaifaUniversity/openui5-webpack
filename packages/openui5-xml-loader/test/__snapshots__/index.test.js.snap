// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XML Loader should handle a fragment 1`] = `
"
      sap.ui.requireSync(\\"sap/m/Button\\");

      var parser = new DOMParser();
      var xml = parser.parseFromString(\\"<core:FragmentDefinition xmlns:core=\\\\\\"sap.ui.core\\\\\\" xmlns=\\\\\\"sap.m\\\\\\"><Button/></core:FragmentDefinition>\\", \\"text/xml\\");
      module.exports = xml;
    "
`;

exports[`XML Loader should handle custom name for root element in view 1`] = `
"
      sap.ui.requireSync(\\"sap/m/Button\\");

      var parser = new DOMParser();
      var xml = parser.parseFromString(\\"<m:View xmlns:m=\\\\\\"sap.ui.core.mvc\\\\\\" xmlns=\\\\\\"sap.m\\\\\\"><Button /></m:View>\\", \\"text/xml\\");
      module.exports = xml;
    "
`;

exports[`XML Loader should handle custom rootPaths 1`] = `
"
      sap.ui.requireSync(\\"./fixtures/custom/TestControl\\");

      var parser = new DOMParser();
      var xml = parser.parseFromString(\\"<mvc:View xmlns:mvc=\\\\\\"sap.ui.core.mvc\\\\\\" xmlns:c=\\\\\\"my.custom.namespace\\\\\\">\\\\n  <c:TestControl />\\\\n</mvc:View>\\", \\"text/xml\\");
      module.exports = xml;
    "
`;
